### ISSUE TITLE
Theming Testing and Repair

### ISSUE THREAD

**Issue by rbayly** [xpologistics/Griffon#65]
Using the New Long Beard Preview App, Setup a  Button and then work through setting up the theming to ensure it works and to walk through how it needs to be setup and configured. This will also allow us to properly set the dev package template with the proper css/theme implementation.

**Comment by rbayly** [comment]
updated ticket with Boris for the theming test monday

**Comment by b-d-k** [comment]
Bunch of comments
---

**General note:** Longbeard is still a moving target, making it difficult to create even a POC.

**Update:** My suggestion to make the material styles part of the repository to make it editable directly received positive feedback from Ryan and Suzanne. 

**Requirements and discussion:** The next step would be to move the Material code into the appropriate space. Since a lot of css code is tied to it's component (i.e. button.d.ts, button-base.scss, button-theme.scss etc.), I am assuming that the entire material lib needs to be moved into the longbeard app. Maybe that is a good moment to decide which components are going to be used in the XPO internal lib?

Also needed is a place for the sass files outside of the material lib. Depending on the planned structure that could either just be a few files that define general rules (with component specific styles in each component folder), or a whole bunch of folders and sass files for the entire lib (excluding of course what is already within the Material lib).

**Comment by rbayly** [comment]
Update on Theming .


1.    Components.

A.    Components shouldn’t be kept in the longbeard site, as we would utilize those as core items in the library, this is where @xpo/core comes in, core is the central library dependency for every library we make, this allows us the ability to write once and run anywhere, since MD components and even the css architecture is central, and then we propagate it out to the libraries this would be the ideal place for it.

B.    Here is a good basic overview of my thoughts on what core would do. And how we would use it. Again this is not set in stone and is a moving target so to speak, but this allows us to build the foundation for the entire component library architecture and set the pace and stage for how we will handle global  functions and variables as well as source components.

Here is something that helped me when I was first trying to decide how to approach this, I came up with this graphic and it helped me wrap my head around the library setup.
I have updated it since we started talking about dictating the css classes in the library to be able to extend them in the apps.


The material styles should again be part of the core, allowing us to set them in one library and them propogate and utilize them in multiple. This also helps us maintain them as a single source items.

We shouldn’t need to move the components into the library or the core, we can utilize them as a dependency and then assign css classes to them inside core, thus allowing us the ability to create our own theme structure without the need to copy components.











-- Liam Bayly
Innovation | Node Development Group
Team Lead

XPOLogistics
13777 Ballantyne Corporate Place
Charlotte, NC 28277 USA

Office: (980) 267-3155
Cell:   (240) 644-4901
Email:   liam.bayly@xpo.com<mailto:ray.bayly@xpo.com?subject=Returning%20Email>
Slack - @LiamBayly<slack://>
